examples/vector_database/scripts/build_vectordb.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import base64
+import glob
+import logging
+import os
+import pickle
+import shutil
+import tempfile
+
+import chromadb
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def list_local_parquet_files(mount_path: str, prefix: str) -> list:
+    """List all parquet files in the mounted S3 directory."""
+    search_path = os.path.join(mount_path, prefix, "**/*.parquet")
+    parquet_files = glob.glob(search_path, recursive=True)
+    return parquet_files
+
+
+def read_parquet_file(file_path: str) -> pd.DataFrame:
+    """Read a parquet file into a pandas DataFrame."""
+    return pd.read_parquet(file_path)
+
+
+def process_batch(collection, batch_df):
+    """Process a batch of data and add it to the ChromaDB collection."""
+    # Extract data from DataFrame and unpack the pickled data
+    ids = [str(idx) for idx in batch_df['idx']]
+
+    # Unpack the pickled data to get images and embeddings
+    unpacked_data = [pickle.loads(row) for row in batch_df['output']]
+    # Each row now contains (image_base64, embedding)
+    images_base64, embeddings = zip(*unpacked_data)
+
+    embeddings = [embedding for embedding in embeddings]
+    # Store both image data and embeddings in metadata
+    metadatas = [{'image_base64': img} for img in images_base64]
+
+    # Add to collection
+    collection.add(ids=ids, embeddings=embeddings, metadatas=metadatas)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Build ChromaDB from mounted S3 parquet files')
+    parser.add_argument('--collection-name',
+                        type=str,
+                        default='clip_embeddings',
+                        help='ChromaDB collection name')
+    parser.add_argument('--persist-dir',
+                        type=str,
+                        default='/vectordb/chroma',
+                        help='Directory to persist ChromaDB')
+    parser.add_argument('--batch-size',
+                        type=int,
+                        default=1000,
+                        help='Batch size for processing')
```

Comment:
Maybe mention why we need a limit here, e.g. entire batch (images + embeddings) has to fit in memory